hack/deploy-karmada.sh

```diff
@@ -152,19 +152,22 @@ util::create_signing_certkey "" "${CERT_DIR}" ca karmada '"client auth","server
 util::create_signing_certkey "" "${CERT_DIR}" front-proxy-ca front-proxy-ca '"client auth","server auth"'
 util::create_signing_certkey "" "${CERT_DIR}" etcd-ca etcd-ca '"client auth","server auth"'
 # signs a certificate
-util::create_certkey "" "${CERT_DIR}" "ca" karmada system:admin "system:masters" kubernetes.default.svc "*.etcd.karmada-system.svc.cluster.local" "*.karmada-system.svc.cluster.local" "*.karmada-system.svc" "localhost" "127.0.0.1" "${interpreter_webhook_example_service_external_ip_address}"
-util::create_certkey "" "${CERT_DIR}" "ca" apiserver karmada-apiserver "" "*.etcd.karmada-system.svc.cluster.local" "*.karmada-system.svc.cluster.local" "*.karmada-system.svc" "localhost" "127.0.0.1" $(util::get_apiserver_ip_from_kubeconfig "${HOST_CLUSTER_NAME}")
-util::create_certkey "" "${CERT_DIR}" "front-proxy-ca" front-proxy-client front-proxy-client "" kubernetes.default.svc "*.etcd.karmada-system.svc.cluster.local" "*.karmada-system.svc.cluster.local" "*.karmada-system.svc" "localhost" "127.0.0.1"
-util::create_certkey "" "${CERT_DIR}" "etcd-ca" etcd-server etcd-server "" kubernetes.default.svc "*.etcd.karmada-system.svc.cluster.local" "*.karmada-system.svc.cluster.local" "*.karmada-system.svc" "localhost" "127.0.0.1"
-util::create_certkey "" "${CERT_DIR}" "etcd-ca" etcd-client etcd-client "" "*.etcd.karmada-system.svc.cluster.local" "*.karmada-system.svc.cluster.local" "*.karmada-system.svc" "localhost" "127.0.0.1"
+karmadaAltNames=(kubernetes.default.svc "*.etcd.karmada-system.svc.cluster.local" "*.karmada-system.svc.cluster.local" "*.karmada-system.svc" "localhost" "127.0.0.1" $(util::get_apiserver_ip_from_kubeconfig "${HOST_CLUSTER_NAME}") "${interpreter_webhook_example_service_external_ip_address}")
+# create karmada-client.crt (CN=system:admin, O=system:masters), mainly used to construct kubeconfig for other components to access karmada-apiserver
+util::create_certkey "" "${CERT_DIR}" "ca" karmada-client system:admin system:masters "${karmadaAltNames[@]}"
+# create karmada-server.crt (CN=karmada-server, O=''), mainly used as server certificate for karmada components, such as karmada-apiserver, karmada-search, karmada-metrics-adapter, etc.
+util::create_certkey "" "${CERT_DIR}" "ca" karmada-server karmada-server "" "${karmadaAltNames[@]}"
+util::create_certkey "" "${CERT_DIR}" "front-proxy-ca" front-proxy-client front-proxy-client "" "${karmadaAltNames[@]}"
+util::create_certkey "" "${CERT_DIR}" "etcd-ca" etcd-server etcd-server "" "${karmadaAltNames[@]}"
+util::create_certkey "" "${CERT_DIR}" "etcd-ca" etcd-client etcd-client "" "${karmadaAltNames[@]}"
```

Comment:
Is it necessary to unify all certificates' altNames?

Reply:
I'm ok with this. We could also extract the common parts of all certificates as commonAltNames and then add altNames to each certificate as needed. Of course, this is just a suggestion.

Reply:
```shell
commanAltNames=(
  "*.etcd.karmada-system.svc.cluster.local"
  "*.karmada-system.svc.cluster.local"
  "*.karmada-system.svc"
  "localhost"
  "127.0.0.1"
)

apiserverAltName=("${commanAltNames[@]}" kubernetes.default.svc "xxx.xxx.xxxx")
```